hack/generator/pkg/codegen/pipeline/flatten_properties.go

```diff
@@ -67,17 +67,28 @@ func makeFlatteningVisitor(defs astmodel.Types) astmodel.TypeVisitor {
 
 			return result, nil
 		},
-		VisitFlaggedType: func(this *astmodel.TypeVisitor, it *astmodel.FlaggedType, ctx interface{}) (astmodel.Type, error) {
-			// skip ARM types, do not flatten
-			if it.HasFlag(astmodel.ARMFlag) {
```

Comment:
Seems sorta weird to visit ARM types and then unflatten them later? Why bother visiting them at all?

Reply:
Instead of skipping them here I remove the `flatten: true` on any properties in an ARM type. This lets us capture if we haven't handled any `flatten: true` properties later.

Reply:
Ahh somehow I was thinking that `removeFlatten` was actually unflattening them - I see it now. Resolving this.